src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository)
+    {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        if (!$this->thereIsADiscount($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $mostRecentLogEntry = $this->getMostRecentLogEntry($channelPricing);
+
+        $logEntryWithTheLowestPriceInPeriod = $this
+            ->channelPricingLogEntryRepository
+            ->findOneWithTheLowestPriceInPeriod($mostRecentLogEntry, self::PERIOD_IN_DAYS)
+        ;
+
+        if ($logEntryWithTheLowestPriceInPeriod !== null) {
+            $channelPricing->setLowestPriceBeforeDiscount($logEntryWithTheLowestPriceInPeriod->getPrice());
+
+            return;
+        }
+
+        $lowestPriceBeyondPeriod = $this
+            ->channelPricingLogEntryRepository
+            ->findTheMostRecentOneBeyondPeriod($mostRecentLogEntry, self::PERIOD_IN_DAYS)
+        ;
+
+        if ($lowestPriceBeyondPeriod === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        if ($lowestPriceBeyondPeriod->getOriginalPrice() === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
```

Comment:
In all of these if statements you do the same thing and if I understand correctly when the `$logEntryWithTheLowestPriceInPeriod` is `null`, we can safely set `lowestPriceBeforeDiscount` to null. So the code below should work as well:
```suggestion
        $channelPricing->setLowestPriceBeforeDiscount(null);
```

Reply:
But if not, you can only merge this conditions into one:
```suggestion
        $lowestPriceBeyondPeriod = $this
            ->channelPricingLogEntryRepository
            ->findTheMostRecentOneBeyondPeriod($mostRecentLogEntry, self::PERIOD_IN_DAYS)
        ;

        if ($lowestPriceBeyondPeriod === null || $lowestPriceBeyondPeriod->getOriginalPrice() === null) {
            $channelPricing->setLowestPriceBeforeDiscount(null);

            return;
        }
```